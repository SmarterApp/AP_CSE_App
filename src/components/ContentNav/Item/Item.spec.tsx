import React from 'react';
import { mount } from 'enzyme';

import { Item } from '.';

describe('Item', () => {
  it('should be active', () => {
    const itemActive = mount(<Item name={'Test Name'} subItems={[]} active={true} />);

    itemActive.simulate('click');
    expect(itemActive).toMatchSnapshot();

    itemActive.unmount();
  });
});
