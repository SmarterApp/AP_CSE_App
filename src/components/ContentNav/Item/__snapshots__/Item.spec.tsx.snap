// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should be active 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Unknown
    active={true}
    name="Test Name"
    subItems={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": true,
      "name": "Test Name",
      "subItems": Array [],
    },
    "ref": null,
    "rendered": Object {
      "instance": <li
        role="menuitem"
      >
        <div
          class="active"
        >
          <div
            class="chevron"
          />
          <div
            class="item-content"
          >
            Test Name
          </div>
        </div>
        <ul
          class="sub-list "
        />
        <style>
          
        * {
          font-family: PT Serif;
        }
        li {
          position: relative;
          display: flex;
          flex-direction: column;
          font-size: 14px;
          text-indent: 0.25em;
          z-index: 0;
          cursor: pointer;
        }
        ul {
          list-style-type: none;
          padding-left: 0;
          width: auto;
        }
        .sub-list :not(.expanded) {
          display: none;
          height: 0;
          overflow: hidden;
        }
        .item-content {
          display: flex;
          flex-direction: row;
          align-items: center;
          justify-content: space-between;
          margin: 0.5em 0;
          font-weight: bold;
        }
        .chevron {
          display: flex;
          margin-top: 5px;
          margin-left: -30px;
          align-items: center;
          text-indent: 0;
          float: left;
          color: #313335;
          display: inline;
        }
        .active {
          color: #006298;
        }
      
        </style>
      </li>,
      "key": "Test Name",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="active"
          >
            <div
              className="chevron"
            />
            <div
              className="item-content"
            >
              Test Name
            </div>
          </div>,
          <ul
            className="sub-list "
          />,
          <style
            jsx={true}
          >
            
        * {
          font-family: PT Serif;
        }
        li {
          position: relative;
          display: flex;
          flex-direction: column;
          font-size: 14px;
          text-indent: 0.25em;
          z-index: 0;
          cursor: pointer;
        }
        ul {
          list-style-type: none;
          padding-left: 0;
          width: auto;
        }
        .sub-list :not(.expanded) {
          display: none;
          height: 0;
          overflow: hidden;
        }
        .item-content {
          display: flex;
          flex-direction: row;
          align-items: center;
          justify-content: space-between;
          margin: 0.5em 0;
          font-weight: bold;
        }
        .chevron {
          display: flex;
          margin-top: 5px;
          margin-left: -30px;
          align-items: center;
          text-indent: 0;
          float: left;
          color: #313335;
          display: inline;
        }
        .active {
          color: #006298;
        }
      
          </style>,
        ],
        "onClick": [Function],
        "role": "menuitem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="active"
          >
            <div
              class="chevron"
            />
            <div
              class="item-content"
            >
              Test Name
            </div>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="chevron"
              />,
              <div
                className="item-content"
              >
                Test Name
              </div>,
            ],
            "className": "active",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <div
                class="chevron"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "chevron",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "div",
            },
            Object {
              "instance": <div
                class="item-content"
              >
                Test Name
              </div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test Name",
                "className": "item-content",
              },
              "ref": null,
              "rendered": Array [
                "Test Name",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": <ul
            class="sub-list "
          />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "sub-list ",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "ul",
        },
        Object {
          "instance": <style>
            
        * {
          font-family: PT Serif;
        }
        li {
          position: relative;
          display: flex;
          flex-direction: column;
          font-size: 14px;
          text-indent: 0.25em;
          z-index: 0;
          cursor: pointer;
        }
        ul {
          list-style-type: none;
          padding-left: 0;
          width: auto;
        }
        .sub-list :not(.expanded) {
          display: none;
          height: 0;
          overflow: hidden;
        }
        .item-content {
          display: flex;
          flex-direction: row;
          align-items: center;
          justify-content: space-between;
          margin: 0.5em 0;
          font-weight: bold;
        }
        .chevron {
          display: flex;
          margin-top: 5px;
          margin-left: -30px;
          align-items: center;
          text-indent: 0;
          float: left;
          color: #313335;
          display: inline;
        }
        .active {
          color: #006298;
        }
      
          </style>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "
        * {
          font-family: PT Serif;
        }
        li {
          position: relative;
          display: flex;
          flex-direction: column;
          font-size: 14px;
          text-indent: 0.25em;
          z-index: 0;
          cursor: pointer;
        }
        ul {
          list-style-type: none;
          padding-left: 0;
          width: auto;
        }
        .sub-list :not(.expanded) {
          display: none;
          height: 0;
          overflow: hidden;
        }
        .item-content {
          display: flex;
          flex-direction: row;
          align-items: center;
          justify-content: space-between;
          margin: 0.5em 0;
          font-weight: bold;
        }
        .chevron {
          display: flex;
          margin-top: 5px;
          margin-left: -30px;
          align-items: center;
          text-indent: 0;
          float: left;
          color: #313335;
          display: inline;
        }
        .active {
          color: #006298;
        }
      ",
            "jsx": true,
          },
          "ref": null,
          "rendered": Array [
            "
        * {
          font-family: PT Serif;
        }
        li {
          position: relative;
          display: flex;
          flex-direction: column;
          font-size: 14px;
          text-indent: 0.25em;
          z-index: 0;
          cursor: pointer;
        }
        ul {
          list-style-type: none;
          padding-left: 0;
          width: auto;
        }
        .sub-list :not(.expanded) {
          display: none;
          height: 0;
          overflow: hidden;
        }
        .item-content {
          display: flex;
          flex-direction: row;
          align-items: center;
          justify-content: space-between;
          margin: 0.5em 0;
          font-weight: bold;
        }
        .chevron {
          display: flex;
          margin-top: 5px;
          margin-left: -30px;
          align-items: center;
          text-indent: 0;
          float: left;
          color: #313335;
          display: inline;
        }
        .active {
          color: #006298;
        }
      ",
          ],
          "type": "style",
        },
      ],
      "type": "li",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": true,
        "name": "Test Name",
        "subItems": Array [],
      },
      "ref": null,
      "rendered": Object {
        "instance": <li
          role="menuitem"
        >
          <div
            class="active"
          >
            <div
              class="chevron"
            />
            <div
              class="item-content"
            >
              Test Name
            </div>
          </div>
          <ul
            class="sub-list "
          />
          <style>
            
        * {
          font-family: PT Serif;
        }
        li {
          position: relative;
          display: flex;
          flex-direction: column;
          font-size: 14px;
          text-indent: 0.25em;
          z-index: 0;
          cursor: pointer;
        }
        ul {
          list-style-type: none;
          padding-left: 0;
          width: auto;
        }
        .sub-list :not(.expanded) {
          display: none;
          height: 0;
          overflow: hidden;
        }
        .item-content {
          display: flex;
          flex-direction: row;
          align-items: center;
          justify-content: space-between;
          margin: 0.5em 0;
          font-weight: bold;
        }
        .chevron {
          display: flex;
          margin-top: 5px;
          margin-left: -30px;
          align-items: center;
          text-indent: 0;
          float: left;
          color: #313335;
          display: inline;
        }
        .active {
          color: #006298;
        }
      
          </style>
        </li>,
        "key": "Test Name",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <div
              className="active"
            >
              <div
                className="chevron"
              />
              <div
                className="item-content"
              >
                Test Name
              </div>
            </div>,
            <ul
              className="sub-list "
            />,
            <style
              jsx={true}
            >
              
        * {
          font-family: PT Serif;
        }
        li {
          position: relative;
          display: flex;
          flex-direction: column;
          font-size: 14px;
          text-indent: 0.25em;
          z-index: 0;
          cursor: pointer;
        }
        ul {
          list-style-type: none;
          padding-left: 0;
          width: auto;
        }
        .sub-list :not(.expanded) {
          display: none;
          height: 0;
          overflow: hidden;
        }
        .item-content {
          display: flex;
          flex-direction: row;
          align-items: center;
          justify-content: space-between;
          margin: 0.5em 0;
          font-weight: bold;
        }
        .chevron {
          display: flex;
          margin-top: 5px;
          margin-left: -30px;
          align-items: center;
          text-indent: 0;
          float: left;
          color: #313335;
          display: inline;
        }
        .active {
          color: #006298;
        }
      
            </style>,
          ],
          "onClick": [Function],
          "role": "menuitem",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="active"
            >
              <div
                class="chevron"
              />
              <div
                class="item-content"
              >
                Test Name
              </div>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="chevron"
                />,
                <div
                  className="item-content"
                >
                  Test Name
                </div>,
              ],
              "className": "active",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <div
                  class="chevron"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "chevron",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "div",
              },
              Object {
                "instance": <div
                  class="item-content"
                >
                  Test Name
                </div>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Test Name",
                  "className": "item-content",
                },
                "ref": null,
                "rendered": Array [
                  "Test Name",
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <ul
              class="sub-list "
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "sub-list ",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "ul",
          },
          Object {
            "instance": <style>
              
        * {
          font-family: PT Serif;
        }
        li {
          position: relative;
          display: flex;
          flex-direction: column;
          font-size: 14px;
          text-indent: 0.25em;
          z-index: 0;
          cursor: pointer;
        }
        ul {
          list-style-type: none;
          padding-left: 0;
          width: auto;
        }
        .sub-list :not(.expanded) {
          display: none;
          height: 0;
          overflow: hidden;
        }
        .item-content {
          display: flex;
          flex-direction: row;
          align-items: center;
          justify-content: space-between;
          margin: 0.5em 0;
          font-weight: bold;
        }
        .chevron {
          display: flex;
          margin-top: 5px;
          margin-left: -30px;
          align-items: center;
          text-indent: 0;
          float: left;
          color: #313335;
          display: inline;
        }
        .active {
          color: #006298;
        }
      
            </style>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "
        * {
          font-family: PT Serif;
        }
        li {
          position: relative;
          display: flex;
          flex-direction: column;
          font-size: 14px;
          text-indent: 0.25em;
          z-index: 0;
          cursor: pointer;
        }
        ul {
          list-style-type: none;
          padding-left: 0;
          width: auto;
        }
        .sub-list :not(.expanded) {
          display: none;
          height: 0;
          overflow: hidden;
        }
        .item-content {
          display: flex;
          flex-direction: row;
          align-items: center;
          justify-content: space-between;
          margin: 0.5em 0;
          font-weight: bold;
        }
        .chevron {
          display: flex;
          margin-top: 5px;
          margin-left: -30px;
          align-items: center;
          text-indent: 0;
          float: left;
          color: #313335;
          display: inline;
        }
        .active {
          color: #006298;
        }
      ",
              "jsx": true,
            },
            "ref": null,
            "rendered": Array [
              "
        * {
          font-family: PT Serif;
        }
        li {
          position: relative;
          display: flex;
          flex-direction: column;
          font-size: 14px;
          text-indent: 0.25em;
          z-index: 0;
          cursor: pointer;
        }
        ul {
          list-style-type: none;
          padding-left: 0;
          width: auto;
        }
        .sub-list :not(.expanded) {
          display: none;
          height: 0;
          overflow: hidden;
        }
        .item-content {
          display: flex;
          flex-direction: row;
          align-items: center;
          justify-content: space-between;
          margin: 0.5em 0;
          font-weight: bold;
        }
        .chevron {
          display: flex;
          margin-top: 5px;
          margin-left: -30px;
          align-items: center;
          text-indent: 0;
          float: left;
          color: #313335;
          display: inline;
        }
        .active {
          color: #006298;
        }
      ",
            ],
            "type": "style",
          },
        ],
        "type": "li",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
